hist/hist/src/TAxis.cxx

```diff
@@ -1270,7 +1270,7 @@ void TAxis::UnZoom()
 
    //unzoom object owning this axis
    SetRange(0,0);
-   TH1 *hobj1 = (TH1*)GetParent();
+   TH1 *hobj1 = GetParent()->InheritsFrom(TH1::Class()) ? static_cast<TH1*>(GetParent()) : nullptr;
```

Comment:
why not simply `dynamic_cast<TH1*>(GetParent())`?
Or you also should check that GetParent() is not `nullptr`